### ISSUE TITLE
[Bug]: raft_snapshot_agent_config

### ISSUE THREAD

**Issue by shuhaib-kk** [hashicorp/terraform-provider-vault#2183]
### Terraform Core Version

1.0.11

### Terraform Vault Provider Version

3.25.0

### Vault Server Version

1.15

### Affected Resource(s)

raft_snapshot_agent_config

### Expected Behavior

 aws_secret_access_key  should be fetched

### Actual Behavior

 aws_secret_access_key  is not being passed or fetched from vars.
<img width="867" alt="Screenshot 2024-03-11 at 12 35 55 PM" src="https://github.com/hashicorp/terraform-provider-vault/assets/157594894/a4616eab-b62f-4f90-91dc-045d14c90a16">

### Terraform Configuration Files

resource "vault_raft_snapshot_agent_config" "s3_backups" {
  name             = "s3_daily"
  interval_seconds = 1800 # 30 mins
  retain           = 7
  path_prefix      = "/tf-vault-backup/dev/"
  storage_type     = "aws-s3"
  aws_s3_bucket         = var.s3_backup_bucket
  aws_s3_region         = var.s3_backup_bucket_region
  aws_access_key_id     = var.aws_access_key_id
  aws_secret_access_key = var.aws_secret_access_key
}

### Steps to Reproduce

resource "vault_raft_snapshot_agent_config" "s3_backups" {
  name             = "s3_daily"
  interval_seconds = 1800 # 30 mins
  retain           = 7
  path_prefix      = "/tf-vault-backup/dev/"
  storage_type     = "aws-s3"
  aws_s3_bucket         = var.s3_backup_bucket
  aws_s3_region         = var.s3_backup_bucket_region
  aws_access_key_id     = var.aws_access_key_id
  aws_secret_access_key = var.aws_secret_access_key
}

### References

https://registry.terraform.io/providers/hashicorp/vault/latest/docs/resources/raft_snapshot_agent_config#aws_secret_access_key

### Would you like to implement a fix?

None

**Comment by shuhaib-kk** [comment]
![Uploading Screenshot 2024-03-11 at 12.35.55 PM.png…]()

**Comment by shuhaib-kk** [comment]
<img width="867" alt="Screenshot 2024-03-11 at 12 35 55 PM" src="https://github.com/hashicorp/terraform-provider-vault/assets/157594894/428d1d03-f63c-4cf4-8049-ae2e41e03293">

**Comment by shuhaib-kk** [comment]
![Screenshot 2024-03-11 at 1 00 22 PM](https://github.com/hashicorp/terraform-provider-vault/assets/157594894/02b2c1bf-c54f-40b8-8932-d0117d2908da)

**Comment by fairclothjm** [comment]
@shuhaib-kk Hello, I am sorry you are having trouble. Could you please from an error output or error log? It is hard to say exactly what is happening without any error information. The second screenshot does indicate that the automated snapshots config was deleted out-of-band. Did you delete the `s3_daily` config in Vault outside of terraform?

**Comment by shuhaib-kk** [comment]
@fairclothjm Hi, the issue is even though am passing aws_secret_key_id argument using variable.
it is not getting configured on vault side. i also checked in state file where it is coming as 
`"aws_secret_access_key": "",` but am able to see aws_access_key_id in the state file.

**Comment by fairclothjm** [comment]
Thanks @shuhaib-kk! It doesn't look like the `vault_raft_snapshot_agent_config` will pickup any environment variables. Please make sure `aws_secret_access_key` is set in the config.

**Comment by shuhaib-kk** [comment]
@fairclothjm you mean configure out of terraform?

**Comment by fairclothjm** [comment]
@shuhaib-kk Thanks, can you confirm that `var.aws_secret_access_key` is being populated with your variable as expected?

**Comment by shuhaib-kk** [comment]
@fairclothjm yes, var.aws_secret_access_key is getting populated as expected and it is also visible in state file.
